src/pages/Waiting/index.js

```diff
@@ -1,31 +1,72 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
-
+import { useHistory } from "react-router-dom";
+import socket from "../../components/utils/socket";
 import Team from "../../components/Team";
 
 export default function WaitingRoom() {
   const [isReady, setIsReady] = useState(false);
   const [team, setTeam] = useState(0);
+  const [canStart, setCanStart] = useState(false);
+
+  const channelId = 1;
+  const roomId = 1;
+  const roomName = `${channelId}-${roomId}`;
+
+  const history = useHistory();
+  const handleOnClickStart = () => {
+    history.push(`/game/:${roomName}-team${team}`);
+  };
+
+  useEffect(() => {
+    socket.on("connect", () => {
+      socket.emit("join", roomName);
+    });
+  }, []);
 
   const handleReady = () => {
-    setIsReady(!isReady);
-    if (!isReady) {
-      const myTeam = team;
+    if (window.confirm(`${team}팀으로 결정하시겠습니까?`)) {
+      setIsReady(!isReady);
+      socket.emit("select_team", channelId, roomId, roomName, team);
+      socket.on("can_start", (c) => {
+        setCanStart(c);
+      });
     }
   };
 
   return (
     <Waiting>
       <Header>팀 선택</Header>
-      <Teams htmlFor="team">
+      <Teams htmlFor="team" style={{ display: isReady ? "none" : "grid" }}>
         {[1, 2, 3, 4].map((it) => (
-          <Team id={it} key={`team_${it}`} setTeam={setTeam} setIsReady={setIsReady} />
+          <Team id={it} key={`team_${it}`} setTeam={setTeam} isReady={isReady} handleReady={handleReady} />
         ))}
       </Teams>
-      <Button style={{ backgroundColor: isReady ? "#f2aeae" : "#e0dede" }} onClick={handleReady}>
-        {isReady ? "선택완료" : "선택하기"}
+      <Button
+        style={{ backgroundColor: isReady ? "#f2aeae" : "#e0dede", display: isReady ? "none" : "inline" }}
+        onClick={handleReady}
+      >
+        선택하기
       </Button>
-      <p style={{ display: isReady ? "block" : "none" }}>{team}팀을 선택했습니다</p>
+      <div style={{ display: isReady ? "block" : "none", margin: "10px" }}>
+        <strong style={{ display: "block", fontSize: "18px", fontWeight: "700", margin: "20px" }}>
+          {team}팀을 선택했습니다
+        </strong>
+        {canStart ? (
+          <div>
+            모든 팀의 선택이 완료되었습니다
+            <br /> 게임을 시작해봅시다 <br />
+          </div>
+        ) : (
+          <div>
+            다른 팀의 접속을 기다리는 중입니다
+            <br /> 잠시만 기다려주세요 <br />
+          </div>
+        )}
+      </div>
+      <StartBtn style={{ display: isReady && canStart ? "inline-block" : "none" }} onClick={handleOnClickStart}>
```

Comment:
원래 함수명대로 하신다고 하시면, onHandleClickStart이 좀더 나을 것으로 보이고, 개인적으로는 아래와 같이 함수명을 제안해봅니다.
```suggestion
      <StartBtn style={{ display: isReady && canStart ? "inline-block" : "none" }} onClick={onHandleStart}>
```

Reply:
handleStart로 수정해볼게요😊